site/content/integrate/apps/functionality/kv-store.md

```diff
@@ -49,22 +132,41 @@ myData := NestedData{
         Id: "hoan6o9ws7rp5xj7wu9rmysrte",
     },
 }
+// Create an instance of the REST API client, acting as a bot (vs. a user)
+client := appclient.AsBot(callRequest.Context)
+// Store the `myData` struct in the key `my-key`
 valueChanged, err := client.KVSet("my-app", "my-key", myData)
 if err != nil {
     // handle the error
 }
 ```
 
-The stored data can then be retrieved as desired:
+Retrieving nested data is the same as any other data:
 
 ```go
 // Create an empty struct to hold the data
 myData := NestedData{
     Team: TeamData{},
 }
-// Retrieve the value of the key into `myData`
+// Create an instance of the REST API client, acting as a bot (vs. a user)
+client := appclient.AsBot(callRequest.Context)
+// Retrieve the value of the key `my-key` into `myData`
 err := client.KVGet("my-app", "my-key", &myData)
```

Comment:
This example shows a more complex data type, so maybe this already handles the explanation of my previous comment